res/css/structures/_TabbedView.scss

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright 2017 Travis Ralston
+Copyright 2019 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+.mx_TabbedView {
+  margin: 0;
+  padding: 0;
+  display: flex;
+  width: 100%;
+  height: 100%;
+}
+
+.mx_TabbedView_tabLabels {
+  width: 136px;
+  height: 100%;
+  color: $tab-label-fg-color;
+}
+
+.mx_TabbedView_tabLabel {
+  vertical-align: text-top;
+  cursor: pointer;
+  display: block;
+  border-radius: 3px;
+  font-size: 12px;
+  font-weight: 600;
+  height: 20px;
+  margin-bottom: 6px;
+}
+
+.mx_TabbedView_tabLabel_active {
+  background-color: $tab-label-active-bg-color;
+  color: $tab-label-active-fg-color;
+}
+
+// TODO: Remove temporary hack alongside "visit old settings" tab
+.mx_TabbedView_tabLabel_TEMP_HACK {
+  background-color: orange;
+}
+
+.mx_TabbedView_tabLabel_icon {
+  width: 14px;
+  height: 14px;
+  margin-left: 6px;
+  margin-right: 9px;
+  position: relative;
+}
+
+.mx_TabbedView_tabLabel_icon > .mx_TabbedView_maskedIcon {
+  width: 14px;
+  height: 14px;
+  display: inline-block;
+}
+
+.mx_TabbedView_tabLabel_icon > .mx_TabbedView_maskedIcon:before {
```

Comment:
@bwindels this is how I ended up doing masking for the tabs. The basic structure is listed here to make them fit correctly, and the actual mask is defined elsewhere (eg `.mx_UserSettingsDialog_settingsIcon:before`). The `Tab` constructor takes an element (a `span` with the classname for the mask) and that gets put through to the DOM like so: 

![image](https://user-images.githubusercontent.com/1190097/51505739-a5cd1100-1da5-11e9-9401-c75a11eea7d6.png)

end result is dynamic tab icons which also magically get masked when the active tab changes.

Reply:
The label is a flexbox and that seems to cause problems with the mask, so I ended up shoving the whole thing one level deeper.

Reply:
Something to do with the alignment last I tried. I'll give it another go though.

Reply:
right, so this happened:
![image](https://user-images.githubusercontent.com/1190097/51551533-dce60580-1e2b-11e9-8105-9ab56868912a.png)

I've managed to fix it though - please take another look at this / https://github.com/matrix-org/matrix-react-sdk/pull/2476/commits/0deb210fd89b91b41d147c4a08969ea9acc1b480

Reply:
mask-repeat: no-repeat will fix that